tests/unit/test_charm.py

```diff
@@ -27,560 +29,527 @@ def read_file(path: str) -> str:
         content = f.read()
     return content
 
+@pytest.fixture
+def patch_multus_is_ready() -> Generator[Mock, None, None]:
+    with patch(f"{MULTUS_LIB}.is_ready") as patch_multus_is_ready:
+        yield patch_multus_is_ready
+
+@pytest.fixture
+def patch_multus_is_available() -> Generator[Mock, None, None]:
+    with patch(f"{MULTUS_LIB}.multus_is_available") as patch_multus_is_available:
+        yield patch_multus_is_available
+
+@pytest.fixture
+def patch_publish_gnb_identity() -> Generator[Mock, None, None]:
+    with patch(
+        f"{GNB_IDENTITY_LIB}.publish_gnb_identity_information"
+    ) as patch_publish_gnb_identity:
+        yield patch_publish_gnb_identity
+
+@pytest.fixture
+def patch_k8s_client() -> Generator[Mock, None, None]:
+    with patch("charm.KubernetesServicePatch") as patch_k8s_client:
+        yield patch_k8s_client
+
+@pytest.fixture
+def patch_generic_sync_client() -> Generator[Mock, None, None]:
+    with patch("lightkube.core.client.GenericSyncClient") as patch_generic_sync_client:
+        yield patch_generic_sync_client
+
+@pytest.fixture()
+def harness(patch_k8s_client, patch_generic_sync_client, patch_multus_is_ready):
+    patch_multus_is_ready.return_value = True
+    harness = testing.Harness(GNBSIMOperatorCharm)
+    harness.set_model_name(name=NAMESPACE)
+    harness.begin()
+    yield harness
+    harness.cleanup()
+
+def set_up_active_status_charm(harness):
+    harness.handle_exec("gnbsim", [], result=0)
+    harness.add_storage("config", attach=True)
+    harness.set_can_connect(container="gnbsim", val=True)
+    set_n2_relation_data(harness)
+    harness.evaluate_status()
+    assert harness.charm.unit.status == ActiveStatus()
+
+def create_n2_relation(harness) -> int:
+    """Create a relation between gnbsim and AMF.
 
-class TestCharm(unittest.TestCase):
```

Comment:
I don't know that we should get rid of the fact that all the unit tests for the charm class are in a charm test class. This is quite useful for separating unit tests when we have multiple classes.

For instance, we can go with something like this:

```python

class TestCharm:

    patcher_lightkube_client = patch("lightkube.core.client.GenericSyncClient")
    patcher_k8s_service_patch = patch("charm.KubernetesServicePatch")
    patcher_multus = patch(MULTUS_LIB)
    patcher_gnb_identity = patch(GNB_IDENTITY_LIB)

    @pytest.fixture
    def setup(self) -> Generator[Mock, None, None]:
        self.mock_lightkube_client = TestCharm.patcher_lightkube_client.start()
        self.mock_k8s_service_patch = TestCharm.patcher_k8s_service_patch.start()
        self.mock_multus = TestCharm.patcher_multus.start()
        self.mock_gnb_identity = TestCharm.patcher_gnb_identity.start()
    
    @pytest.fixture()
    def harness(self, setup):
        harness = testing.Harness(GNBSIMOperatorCharm)
        harness.set_model_name(name=NAMESPACE)
        harness.begin()
        yield harness
        harness.cleanup()

    def set_up_active_status_charm(self, harness):
        harness.handle_exec("gnbsim", [], result=0)
        harness.add_storage("config", attach=True)
        harness.set_can_connect(container="gnbsim", val=True)
        self.set_n2_relation_data(harness)
        harness.evaluate_status()
        assert harness.charm.unit.status == ActiveStatus()

    def create_n2_relation(self, harness) -> int:
        """Create a relation between gnbsim and AMF.

        Returns:
            int: Id of the created relation
        """
        amf_relation_id = harness.add_relation(relation_name="fiveg-n2", remote_app="amf")
        harness.add_relation_unit(relation_id=amf_relation_id, remote_unit_name="amf/0")
        return amf_relation_id

    def set_n2_relation_data(self, harness) -> int:
        """Create the N2 relation, set the relation data in the n2 relation and return its ID.

        Returns:
            int: ID of the created relation
        """
        amf_relation_id = self.create_n2_relation(harness)
        harness.update_relation_data(
            relation_id=amf_relation_id,
            app_or_unit="amf",
            key_values={
                "amf_hostname": "amf",
                "amf_port": "38412",
                "amf_ip_address": "1.1.1.1",
            },
        )
        return amf_relation_id

    @pytest.mark.parametrize("config_param", [("usim-opc"),
                                            ("gnb-ip-address"),
                                            ("icmp-packet-destination"),
                                            ("imsi"),
                                            ("mcc"),
                                            ("mnc"),
                                            ("usim-key"),
                                            ("usim-sequence-number"),
                                            ("sd"),
                                            ("tac"),
                                            ("upf-subnet"),
                                            ("upf-gateway")])
    def test_given_invalid_config_when_config_changed_then_status_is_blocked(self, harness, config_param):
        harness.update_config(key_values={config_param: ""})
        harness.evaluate_status()

        assert harness.charm.unit.status == BlockedStatus(
            f"Configurations are invalid: ['{config_param}']"
        )
```